### PR TITLE
Upgrade nbconvert

### PR DESCRIPTION
**Pull request by martinRenou**
Revive https://github.com/jupyter/nbviewer/pull/1003 and merge with https://github.com/jupyter/nbviewer/pull/1014

closes #1003 
closes #1014 
closes #959

### CODE REVIEW

**Comment by minrk** [comment]
@martinRenou this mostly works now, but the lab template's font-awesome 5 is conflicting with the font-awesome 4 used on the nbviewer page, erasing the icons:

Screen Shot 2022-08-22 at 14 47 05
 
Not sure if there's a simple fix for that.

**Comment by martinRenou** [comment]
Thanks a lot for pushing on this @minrk !

> but the lab template's font-awesome 5 is conflicting with the font-awesome 4 used on the nbviewer page, erasing the icons:

In Voila we do the following, maybe that will fix the nbviewer case:

```
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@^5/css/all.min.css" type="text/css" />
<link rel="stylesheet" href="https://cdn.jsdelivr.net/npm/@fortawesome/fontawesome-free@^5/css/v4-shims.min.css" type="text/css" />
```

I am not sure where to add this though. I wonder if this should go directly in nbconvert.

**Comment by minrk** [comment]
We should update to fa5, but this backward-incompatibility is definitely an issue. The shims do work in layout.html (after our own css), but result in a different size than before:

before:
Screen Shot 2022-08-24 at 12 39 30

after:
Screen Shot 2022-08-24 at 12 39 27

**Comment by Carreau** [comment]
The navbar seem to be missing visually as well.
Screen Shot 2022-10-31 at 09 31 59

File: nbviewer/templates/notebook.html
Comment (Carreau) [inline nbviewer/templates/notebook.html:117]: DEBUG ?

**Comment by Carreau** [comment]
A huge chink of the css issue seem to be with

```
  {% block ipywidgets %}
    <!-- block ipywidgets notebook.html -->
    <script>
      (function() {
        function addWidgetsRenderer() {
          var mimeElement = document.querySelector('script[type="application/vnd.jupyter.widget-view+json"]');
          var scriptElement = document.createElement('script');
          var widgetRendererSrc = '{{ ipywidgets_base_url }}@jupyter-widgets/html-manager@{{ jupyter_widgets_html_manager_version }}/dist/embed-amd.js';
          var widgetState;

          try {
            widgetState = mimeElement && JSON.parse(mimeElement.innerHTML);

            if (widgetState && (widgetState.version_major < 2 || !widgetState.version_major)) {
              widgetRendererSrc = '{{ ipywidgets_base_url }}jupyter-js-widgets@{{ jupyter_js_widgets_version }}/dist/embed.js';
            }
          } catch(e) {}

          scriptElement.src = widgetRendererSrc;
          document.body.appendChild(scriptElement);
        }

        document.addEventListener('DOMContentLoaded', addWidgetsRenderer);
      }());
    </script>
    <!-- end block ipywidgets notebook.html -->
  {% endblock ipywidgets %}
```

That dynamically inject css.

File: nbviewer/templates/notebook.html
Comment (minrk) [reply nbviewer/templates/notebook.html]: yes, debugging

**Comment by minrk** [comment]
@Carreau thanks for picking this up!

File: README.md
Comment (Carreau) [inline README.md:141]: I think listening on 0.0.0.0 by default is a security problem. Well, here on macos, it present me from doing so without root access so It's ok-ish, but i'd like to fix it.

**Review by Carreau — CHANGES_REQUESTED** [review]
Slides are also quite broken. 

For example you can compare with  https://nbviewer.org/format/slides/github/astrojuanlu/talk-dataframes/blob/main/slides.ipynb

**Comment by Carreau** [comment]
Most of the reveal issue seem to be coming down to font-size being way to big. 

Something like the following in the right place make it bearable.
```
div.reveal {
    font-size: 10px;

}
```

Now we do seem to have 2 sets of reveal controls on the page if you look carefully: 

Screen Shot 2022-10-31 at 16 15 10

**Comment by Carreau** [comment]
So there are indeed 2 reveal. One done by nbconvert, and one by nbviewer. 

Really this is crazy that we are embedding a full nbconvert with `body|safe`, that even include DOCTYPE ????. And of course the Reveal from nbconvert has no hooks, so we can't listen to the slide change event to hide the the nbviewer header.

The new nbconverts are really backward in term of reusing the templates...

**Review by Carreau — APPROVED** [review]

File: nbviewer/providers/github/handlers.py
Comment (Carreau) [inline nbviewer/providers/github/handlers.py:160]:
```suggestion
```

File: nbviewer/templates/formats/slides.html
Comment (Carreau) [inline nbviewer/templates/formats/slides.html:86]: That is the part we can't do without changes to nbviewer.

**Review by Carreau — APPROVED** [review]
I think that's about as far as I we can reasonably take it in a single PR. It works, but there is still alot of work to do.

**Comment by Carreau** [comment]
I think we should leapfrog nbconvert 6.x and go directly to 7.x, I'm going to create a 1.x branch that is the current main, and suggest main become 2.x.

**Comment by minrk** [comment]
If that makes it easier, by all means!

**Comment by Carreau** [comment]
> If that makes it easier, by all means!

It's more on the communication side, there will be likely so much change between 1.x and main, that it does make sens to bump major version number. This also let us re-do a 1.x at one point if ever necessary. 

I've push a 1.x branch, and I'm going to merge this and keep iterating, once this is merged, main should not be considered stable as a number of notebook likely won't render yet, and reveal is still partially broken .

**Comment by martinRenou** [comment]
Thanks a lot for pushing on this!

**Comment by mgeier** [comment]
Would it be possible to get another `nbconvert` upgrade on https://nbviewer.org/?

I think the latest release should fix an issue where SVG images are not displayed correctly: https://github.com/jupyter/nbconvert/issues/1863.
See also #1037.

**Pull request merged**
